make_mock_json.py

```diff
@@ -0,0 +1,310 @@
+2017-06-23 17:18:41,791 platform:x86_64 - atomic_reactor.foo - INFO - 2017-06-23 17:18:41,400 platform:- atomic_reactor.foo -  DEBUG - this is from a worker build
+2017-06-23 05:04:51,334 platform:x86_64 - atomic_reactor.plugins.orchestrate_build - INFO - "ContainersPaused": 0,
+2017-06-23 17:18:41,791 - I really like bacon
+""")
+BASE_BUILD_LOG = u"   líne 1   \n".encode('utf-8')
+MOCK_CONFIG_MAP = {
+    "apiVersion": "v1",
+    "data": {
+        "special.how": "\"very\"",
+        "special.type": "{\"quark\": \"charm\"}",
+        "config.yaml": "version:1",
+        "config.yml": "version:2",
+        "config.ymlll": "{\"version\":3}",
+        "config.json": "{\"version\":4}"
+    },
+    "kind": "ConfigMap",
+    "metadata": {
+        "creationTimestamp": "2017-06-14T16:49:47Z",
+        "name": "special-config",
+        "namespace": "osbs-qa01",
+        "resourceVersion": "117005745",
+        "selfLink": "/api/v1/namespaces/osbs-qa01/configmaps/special-config",
+        "uid": "797c8350-5121-11e7-9ce0-0efeb5d2209c"
+    }
+}
+DEFAULT_DIR = "tests/mock_jsons"
+
+
+def canonize_data(build_data, build_name=TEST_BUILD, build_status=None):
+    build_data["status"]["config"] = {
+        "kind": "BuildConfig",
+        "name": build_name,
+        "namespace": "default",
+    }
+    build_data["metadata"]["name"] = build_name
+    if build_status:
+        build_data["status"]["phase"] = build_status
+        try:
+            del build_data["status"]["logSnippet"]
+            del build_data["status"]["message"]
+        except KeyError:
+            pass
+    return build_data
+
+
+def find_or_make_dir(dir_name):
+    try:
+        os.stat(dir_name)
+    except OSError:
+        try:
+            os.makedirs(dir_name)
+        except OSError:
+            print("cannot open or create {}".format(dir_name))
+            raise
+
+
+class Mock_Creator(object):
+    def __init__(self):
+        parser = argparse.ArgumentParser(description="osbs test harness mock JSON creator")
+
+        parser.add_argument("--config", action='store', metavar="PATH",
+                            help="path to configuration file", default=DEFAULT_CONFIGURATION_FILE)
+        parser.add_argument("--instance", "-i", action='store', metavar="SECTION_NAME",
+                            help="section within config for requested instance",
+                            default="stage")
+        parser.add_argument("--user", action='store',
+                            help="name of user to use for Basic Authentication in OSBS")
+        parser.add_argument("--password", action='store',
+                            help="password to use for Basic Authentication in OSBS")
+        parser.add_argument("--mock-dir", metavar="DIR", action="store", default=DEFAULT_DIR,
+                            help="mock JSON responses are stored in DIR")
+        parser.add_argument("--os_version", metavar="OS_VER", action="store", default=OS_VERSION,
```

Comment:
Can we name it `"--os-version"` i.e. all dashes, no underscore?

Reply:
Sure.